src/material-experimental/mdc-slide-toggle/slide-toggle.ts

```diff
@@ -15,11 +59,250 @@ import {ChangeDetectionStrategy, Component, ViewEncapsulation} from '@angular/co
+    foundation.setDisabled(this.disabled);
+    foundation.setChecked(this.checked);
+  }
+
+  ngOnDestroy() {
+    if (this._foundation) {
+      this._foundation.destroy();
+    }
+  }
+
+  /** Method being called whenever the underlying input emits a change event. */
+  _onChangeEvent(event: Event) {
+    // We always have to stop propagation on the change event.
+    // Otherwise the change event, from the input element, will bubble up and
+    // emit its event object to the component's `change` output.
+    event.stopPropagation();
+    this.toggleChange.emit();
+    this._foundation.handleChange(event);
+
+    // When the slide toggle's config disabled toggle change event by setting
+    // `disableToggleValue: true`, the slide toggle's value does not change,
+    // and the checked state of the underlying input needs to be changed back.
+    if (this.defaults.disableToggleValue) {
+      this._inputElement.nativeElement.checked = this.checked;
+      return;
+    }
+
+    // Sync the value from the underlying input element with the component instance.
+    this.checked = this._inputElement.nativeElement.checked;
+
+    // Emit our custom change event only if the underlying input emitted one. This ensures that
+    // there is no change event, when the checked state changes programmatically.
+    this.onChange(this.checked);
+    this.change.emit(new MatSlideToggleChange(this, this.checked));
+  }
+
+  /** Method being called whenever the slide-toggle has been clicked. */
+  _onInputClick(event: Event) {
+    // We have to stop propagation for click events on the visual hidden input element.
+    // By default, when a user clicks on a label element, a generated click event will be
+    // dispatched on the associated input element. Since we are using a label element as our
+    // root container, the click event on the `slide-toggle` will be executed twice.
+    // The real click event will bubble up, and the generated click event also tries to bubble up.
+    // This will lead to multiple click events.
+    // Preventing bubbling for the second event will solve that issue.
+    event.stopPropagation();
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  writeValue(value: any): void {
+    this.checked = !!value;
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  /** Implemented as a part of ControlValueAccessor. */
+  setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+    this._changeDetectorRef.markForCheck();
+  }
+
+  /** Focuses the slide-toggle. */
+  focus(): void {
+    this._inputElement.nativeElement.focus();
+  }
+
+  /** Toggles the checked state of the slide-toggle. */
+  toggle(): void {
+    this.checked = !this.checked;
+    this.onChange(this.checked);
```

Comment:
should we be doing this here? we don't want it for programmatic toggle calls do we?

Reply:
I took this from the existing slide toggle so the behavior stays consistent.

Reply:
I see, can you add a TODO on it? maybe we'll come back to it some day

Reply:
Looking through it again, this is the `onChange` callback from `ControlValueAccessor`. We need to call it for every value change so the CVA stays in sync.